decorator.rb

```diff
@@ -0,0 +1,28 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.size >= 11
+      @nameable.correct_name.slice(0, 10)
+    else
+      @nameable.correct_name
+    end
```

Comment:
- [ ] [OPTIONAL] Instead of using if-else we can use this.
-  `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

Reply:
Thank you reviewer, for your review and suggestion. I'll make that change.